prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -438,27 +451,30 @@ class FHIRRouter(
     }
 
     /**
-     * Takes a [bundle] and [filter] and optionally a [focusResource], evaluates if the bundle passes the filter, or the
-     * opposite if [reverseFilter] is true. If the filter is null or empty, return [defaultResponse].
-     * @return Boolean indicating if the bundle passes the filter or not
-     *         String to use when logging the filter result
+     * Takes a [bundle] and [filter] and optionally a [focusResource], evaluates if the bundle passes all
+     * the filter conditions, or when [reverseFilter] is true, evaluates if the bundle doesn't pass at least one of the
+     * filter conditions.
+     * @param filter Filter that will be evaluated
+     * @param bundle FHIR Bundle that will be evaluated
+     * @param defaultResponse result when there are no filter conditions in the list
+     * @param reverseFilter Optional flag used to reverse the result of the filter evaluation
+     * @param focusResource Starting point for the evaluation, can be [bundle] if checking from root
+     * @return Pair: Boolean indicating if the bundle passes the filter or not
+     *         and String to use when logging the filter result
      */
-    internal fun evaluateFilterCondition(
+    internal fun evaluateFilterConditionAsAnd(
         filter: ReportStreamFilter?,
         bundle: Bundle,
         defaultResponse: Boolean,
         reverseFilter: Boolean = false,
         focusResource: Base = bundle,
     ): Pair<Boolean, String?> {
-        // the filter needs to check all expressions passed in, or if the filter is null or empty it will return the
-        // default response
         if (filter.isNullOrEmpty()) {
             return Pair(defaultResponse, "defaultResponse")
         }
-
+        check(filter.isNotEmpty())
         val failingFilters = mutableListOf<String>()
```

Comment:
Is this check necessary since the if before is checking if filter is null or empty?

Reply:
No it's not, that's leftover from an earlier iteration, we can remove it, good catch.